### DISCUSSION TITLE
Are weights within wildcards supported?

### DISCUSSION THREAD

**Discussion by Kadah** [adieyal/sd-dynamic-prompts#399] [Q&A] [answered]
I  was trying to add weights in a wildcard file. Something like this:
```
0.2::winter
0.5::spring
0.4::summer
0.7::fall
```

Couldn't tell if weights were having any affect on their selection from the limited testing I did, but the prompt is getting the full literal line applied.

Example:
`A photo of a landscape in __season__`
`A photo of a landscape in 0.5::spring`

Not sure if this is a thing. I guessing it might work if using the full `{::|::}` notation instead.

**Comment by adieyal** [comment, marked as answer]
sorry, no. For short lists, it might be easiest to  simply use a variant with weights. There is no clean solution for longer weights but you can try using a yaml file like so:

```
##### colours.yaml #####
colours:
  - |
    {
        0.1:: red,
        0.3:: blue
        0.6::green
        ....
    }
```

**Reply by akx** [reply]
We could add this to structured wildcard files though, I've been thinking about it. :)

**Reply by Kadah** [reply]
Was thinking it might be possible via the yaml support instead. Documentation and examples for yaml files is a bit light at the moment.

**Reply by adieyal** [reply]
have a look at this file - https://github.com/adieyal/sd-dynamic-prompts/blob/main/collections/publicprompts.yaml
It should give you an idea of what's possible
